docs/4.4/api-reference.md

```diff
@@ -11,57 +11,287 @@ Teleport is currently working on documenting our API.
-`auth.crt`, `auth.key`, and `auth.cas` (certificate, private key, and CA certs respectively).
+This should result in three PEM encoded files being generated in the `/certs` directory: `api-admin.crt`, `api-admin.key`, and `api-admin.cas` (certificate, private key, and CA certs respectively).
 
-Note: by default, tctl auth sign produces certificates with a relatively short lifetime.
-For production deployments, the --ttl flag can be used to ensure a more practical
-certificate lifetime.
+!!! Note
+    By default, `tctl auth sign` produces certificates with a relatively short lifetime.
+    For production deployments, the `--ttl` flag can be used to ensure a more practical
+    certificate lifetime.
 
-# gRPC APIs
+## Setup Go Client
+
+Put `client.go` (below) in the same folder as `certs`. Then you can create a `main.go` file and use the client from `connectClient` to run the following examples, or experiment on your own. Your folder structure should look like this:
+
+```
+.
++-- api-admin.yaml
++-- certs
+|   +-- api-admin.cas
+|   +-- api-admin.crt
+|   +-- api-admin.key
++-- client.go
++-- main.go
+```
+
+**client.go**
+```go
+{!examples/go-client/client.go!}
+```
+
+## Roles
+
+[Roles](enterprise/ssh-rbac.md#roles) are used to define what resources and actions a user is allowed/denied access to. It's best to make your role's permissions as stringent as possible.
+
+**Create a new role**
+
+```go
+// create a new auditor role which has very limited permissions
+role, err := services.NewRole("auditor", services.RoleSpecV3{
+  Options: services.RoleOptions{
+    MaxSessionTTL: services.Duration(time.Hour),
+  },
+  Allow: services.RoleConditions{
+    Logins: []string{"auditor"},
+    Rules: []services.Rule{
+      // - resources: ['session']
+      //   verbs: ['list', 'read']
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    // node_labels: '*': '*'
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Retrieve role**
+
+```go
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+```
+
+**Update role**
+
+```go
+// update the auditor role's read rule to not provide access to secrets
+role.SetRules(services.Allow, []services.Rule{
+	services.NewRule(services.KindSession, services.ReadNoSecrets()),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Delete role**
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add remote clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+**Create token**
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a non-random tokenString
```

Comment:
it reads a bit out of place - I think it could be better with an actual commented out example:
```go
tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
  // Token: "custom token value", // Set token explicitly, otherwise it will be generated
  Roles: teleport.Roles{teleport.RoleProxy},
  TTL:   time.Hour,
})
```